src/game/game_gui.cpp

```diff
@@ -0,0 +1,467 @@
+						DrawBoolButton(buttons_left, y + button_y_offset, current_value != 0, editable);
+						SetDParam(idx++, current_value == 0 ? STR_CONFIG_SETTING_OFF : STR_CONFIG_SETTING_ON);
+					} else {
+						if (config_item.complete_labels) {
+							DrawDropDownButton(buttons_left, y + button_y_offset, COLOUR_YELLOW, this->clicked_row == i && clicked_dropdown, editable);
+						} else {
+							DrawArrowButtons(buttons_left, y + button_y_offset, COLOUR_YELLOW, (this->clicked_button == i) ? 1 + (this->clicked_increase != rtl) : 0, editable && current_value > config_item.min_value, editable && current_value < config_item.max_value);
+						}
+						if (config_item.labels != nullptr && config_item.labels->Contains(current_value)) {
+							SetDParam(idx++, STR_JUST_RAW_STRING);
+							SetDParamStr(idx++, config_item.labels->Find(current_value)->second);
+						} else {
+							SetDParam(idx++, STR_JUST_INT);
+							SetDParam(idx++, current_value);
+						}
+					}
+
+					DrawString(text_left, text_right, y + text_y_offset, str, colour);
+					y += this->line_height;
+				}
+				break;
+			}
+		}
+	}
+
+	void OnPaint() override
+	{
+		if (this->closing_dropdown) {
+			this->closing_dropdown = false;
+			this->clicked_dropdown = false;
+		}
+		this->DrawWidgets();
+	}
+
+	void OnClick(Point pt, int widget, int click_count) override
+	{
+		if (widget >= WID_GSC_TEXTFILE && widget < WID_GSC_TEXTFILE + TFT_END) {
+			if (GameConfig::GetConfig() == nullptr) return;
+
+			ShowScriptTextfileWindow((TextfileType)(widget - WID_GSC_TEXTFILE), (CompanyID)OWNER_DEITY);
+			return;
+		}
+
+		switch (widget) {
+			case WID_GSC_GSLIST: {
+				this->InvalidateData();
+				if (click_count > 1 && _game_mode != GM_NORMAL) ShowAIListWindow((CompanyID)OWNER_DEITY);
+				break;
+			}
+
+			case WID_GSC_CHANGE:  // choose other Game Script
+				ShowAIListWindow((CompanyID)OWNER_DEITY);
+				break;
+
+			case WID_GSC_CONTENT_DOWNLOAD:
+				if (!_network_available) {
+					ShowErrorMessage(STR_NETWORK_ERROR_NOTAVAILABLE, INVALID_STRING_ID, WL_ERROR);
+				} else {
+					ShowNetworkContentListWindow(nullptr, CONTENT_TYPE_GAME);
+				}
+				break;
+			case WID_GSC_SETTINGS: {
+				const NWidgetBase* wid = this->GetWidget<NWidgetBase>(WID_GSC_SETTINGS);
+				int num = (pt.y - wid->pos_y) / this->line_height + this->vscroll->GetPosition();
+				if (num >= (int)this->visible_settings.size()) break;
+
+				VisibleSettingsList::const_iterator it = this->visible_settings.begin();
+				for (int i = 0; i < num; i++) it++;
+				const ScriptConfigItem config_item = **it;
+				if (!this->IsEditableItem(config_item)) return;
+
+				if (this->clicked_row != num) {
+					this->CloseChildWindows(WC_QUERY_STRING);
+					HideDropDownMenu(this);
+					this->clicked_row = num;
+					this->clicked_dropdown = false;
+				}
+
+				bool bool_item = (config_item.flags & SCRIPTCONFIG_BOOLEAN) != 0;
+
+				int x = pt.x - wid->pos_x;
+				if (_current_text_dir == TD_RTL) x = wid->current_x - 1 - x;
```

Comment:
This section is copied over from the `AISettingsWindow`, since that window is now embedded into the GS config window. Thus, this `1` is just ported over from that window (See [ai_gui.cpp](https://github.com/OpenTTD/OpenTTD/blob/8c4a70e05ea6ad1387961f96e377a00fac66fdd7/src/ai/ai_gui.cpp#L458)) and I didn't end up changing it.

I was considering looking through the code and also making all the re-used code between both `ai_gui.cpp` and `game_gui.cpp` actually shared instead of duplicated, but decided not to because I wanted to make the changes as minimal as possible for this PR.

Reply:
After some discussion on Discord, it seems that #10114 might make this a non-issue.